### PR TITLE
Parallelize ls tasks

### PR DESCRIPTION
**Pull request by emord**
The LS tasks don't have any dependencies except for agg_ls table, but they were kicking off tasks one by one.

### CODE REVIEW

**Comment by emord** [comment]
@calellowitz thoughts on this PR? I think you said that these may be parallel in the future airflow owrld, but I'm not sure when that is?

File: custom/icds_reports/tasks.py [hunk 1]
```diff
@@ -308,29 +308,40 @@ def move_ucr_data_into_aggregation_tables(date=None, intervals=2, force_citus=Fa
...
-                res_ls_tasks.append(icds_aggregation_task.si(date=calculation_date, func_name='_agg_ls_table', force_citus=force_citus))
-
-                res_awc = chain(icds_aggregation_task.si(date=calculation_date, func_name='_agg_awc_table', force_citus=force_citus),
-                                *res_ls_tasks
-                                ).apply_async()
-
+                res_awc = icds_aggregation_task.si(
+                    date=calculation_date, func_name='_agg_awc_table', force_citus=force_citus
+                ).delay()
+                res_ls = icds_aggregation_task.si(
```
Comment (calellowitz) [inline custom/icds_reports/tasks.py]: I think this depends on agg_awc

**Review by calellowitz — CHANGES_REQUESTED** [review]
One comment, but also noting that im hoping to do the airflow switchover on tuesday or wednesday of next week

File: custom/icds_reports/tasks.py (hunk 1 above)
Comment (emord) [reply custom/icds_reports/tasks.py]: Can you show me where? The only tables I see referenced are three intermediate tables (they only depend on UCR) https://github.com/dimagi/commcare-hq/blob/20983d50bd9d33ae8120a3c2fb6b57b19ff09a0b/custom/icds_reports/utils/aggregation_helpers/monolith/agg_ls_data.py#L9, itself https://github.com/dimagi/commcare-hq/blob/20983d50bd9d33ae8120a3c2fb6b57b19ff09a0b/custom/icds_reports/utils/aggregation_helpers/monolith/agg_ls_data.py#L16, and `awc_location_local` https://github.com/dimagi/commcare-hq/blob/20983d50bd9d33ae8120a3c2fb6b57b19ff09a0b/custom/icds_reports/utils/aggregation_helpers/monolith/agg_ls_data.py#L100

Maybe this means you can also change your airflow to be more parallel? From what I can tell, I think you could even have an LS job totally separate from the AWC aggregations. (There's just a select for `awc_location_local` which should still be query-able even if the AWC agg is running

Comment (Rohit25negi) [reply custom/icds_reports/tasks.py]: This is fine to make the agg_ls independent to agg_awc. Initially when this code was merged it used to depened on agg_awc completion(not for through code but to keep the semantics of indicator right [see here](https://github.com/dimagi/commcare-hq/pull/22458/files#r235128250)).  But[ Later change](https://github.com/dimagi/commcare-hq/pull/22617/files#diff-77bde2e45c4a27b16ce4ef95e6e17a14R33) in the spec of LS dashboard invalidated that requirement and while making this later change I did not think of making it independent here.
@emord

**Review by Rohit25negi — APPROVED** [review]
Looks good to me

**Comment by emord** [comment]
Closing because airflow is being released and this won't be used anymore. @calellowitz may still be useful to parallelize this in airflow?

**Pull request closed without merging**